pyvrp/cpp/DurationSegment.cpp

```diff
@@ -3,19 +3,19 @@
 using pyvrp::Duration;
 using pyvrp::DurationSegment;
 
-Duration DurationSegment::twEarly() const { return twEarly_; }
-
-Duration DurationSegment::twLate() const { return twLate_; }
+Duration DurationSegment::earliestStart() const { return earliestStart_; }
 
 Duration DurationSegment::releaseTime() const { return releaseTime_; }
 
 DurationSegment::DurationSegment(size_t idx, ProblemData::Client const &client)
     : idxFirst_(idx),
       idxLast_(idx),
       duration_(client.serviceDuration),
-      timeWarp_(0),
-      twEarly_(client.twEarly),
-      twLate_(client.twLate),
+      earliestStart_(client.twEarly),
+      latestFinish_(client.twLate > std::numeric_limits<Duration>::max()
+                                        - client.serviceDuration
+                        ? std::numeric_limits<Duration>::max()
+                        : client.twLate + client.serviceDuration),
```

Comment:
I just took a look at this PR in more detail after our Slack discussion. I originally it was only about renaming, but why do we need to reparametrize the duration segment? The net change is one less parameter, but we now have to handle an overflow and the implementation diverges from the user interface.

Reply:
There is no strict need but the current implementation is redundant, i.e. it uses two parameters with a restriction on their combinations so in a sense it is just simpler to make that explicit using a single parameter.

Additionally I think it makes the merge conceptually simpler as its mostly just updating the earliest start and latest finish, which both follow from the earliest start / latest finish from either the first or second segment, depending on which one is more restricting.

Finally (a personal argument) this parameterization is easier to generalize: i.e. with time dependent travel times, time warp is a function of start time while the latest finish is constant, greatly simplifying things.